contents/docs/contribute/coding-conventions/index.md

```diff
@@ -51,8 +51,40 @@ Don’t log sensitive information. Make sure you never log:
 #### Unit tests
 A good unit test should:
 * focus on a single use-case at a time
-* have a minimal set of assertion per test
+* have a minimal set of assertions per test
 * demonstrate every use case. The rule of thumb is: if it can happen, it should be covered
 
 #### Integration tests
 Integration tests should ensure that the feature works end-to-end. They must cover all the important use cases of the feature.
+
+### Frontend 
+
+#### Two layers: Kea -> React
+
+Our frontend webapp is written with [Kea](https://keajs.org/) and [React](https://reactjs.org/) as two separate layers. Kea is used to organise the app's data for rendering (we call this the *data* or *state* layer), and React is used to render the computed state (this is the *view* or *template* layer).
+
+We try to be very explicit about this separation, and avoid local React state wherever possible, with exceptions for the `lib/` folder. Having all our data in one layer makes for code that's easier to [test](https://kea.js.org/docs/guide/testing), and observe. Basically, getting your [data layer](https://kea.js.org/blog/data-first-frontend-revolution) right is hard enough. We aim to not make it harder by constraining your data to a DOM-style hierarchy.
+
+Hence the explicitly in keeping the layers separate.
+
+#### General tips
+
+- Do think data first: get [your mental model of the data flowing through the app](https://acco.io/i-escaped-node) right, and then everything else will be easy.
+- Be practical and pragmatic, yet remember that you're writing code that's expected to stick around for the next 1-3 years, not the next 3 weeks. This means optimise for mid-to-long term maintainability over quick wins that are measured in hours or minutes.
+
+#### Do-s & Don't-s
+
+- Write all new code with TypeScript and proper typing.
+- Write your frontend data handling code first, and write it in a Kea `logic`.
+- Don't use `useState` or `useEffect` to store local state. It's false convenience. Take the extra 3 minutes and change it to a `logic` early on in the development.
+- Logics still have a tiny initialisation cost. Hence this rule doesn't apply to library components in the `lib/` folder, which might be rendered hundreds of times on a page with different sets of data. Still feel free to write a logic for a complicated `lib/` component when needed.
+- Logics are camelcase and start with a lowercase (`dashboardLogic`). React components are camelcase and start with an uppercase (`DashboardMenu`).
+- Name the `.ts` file accoridng to its main export: `.ts` or `DashboardMenu.tsx`
+- Use only named exports (`export const DashboardMenu = () => <div />`).
+- Do not use `default` exports like `export default MyBigComponent`
+- We use regular SCSS files for styling to keep things simple and maintainable.
```

Comment:
Personally I would actually prefer CSS-in-JS over these tons of separate `.scss` files which have to bundled anyway. :D Emotion is fantastic IMO: https://emotion.sh/, here's some production styles: https://github.com/Twixes/hubmeter/blob/master/src/features/Controls/styles.ts

Reply:
I'm happy to rethink this decision eventually... though I do remember going through a lot of pain with emotion and server-side rendering in my last job because `react-select` chose to use it: https://github.com/JedWatson/react-select/issues/2706 -- all the exceptions I had to add for that still keep me up at night.

I've also gone through a few CSS-in-JS approaches myself only to find myself rewriting things to the next framework once community adoption moved on. The same is true for any kind of flavour of the month. CoffeeScript instead of JS gets eventually rewritten. HAML instead of HTML gets eventually rewritten. SASS/Emotion/StyledComponents/Radium/whatever instead of SCSS gets eventually rewritten.

Keeping our stack as standard as possible is a massive productivity win over the long run.